refactor save_data as non member function

ann-ss opened the pull request:
This PR changes the save_data() member functions of torch::jit::mobile::Module which was introduced in #41403 to be the non member function torch::jit::mobile::_save_parameters() (taking a mobile Module as its first argument).

In addition, this PR:
* adds a getter function _ivalue() for the mobile::Module object
* renames torch::jit::mobile::_load_mobile_data() to torch::jit::mobile_load_parameters()
* refactors the import.h header file into import.h and import_data.h

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42045 refactor save_data as non member function**

Differential Revision: [D22766781](https://our.internmc.facebook.com/intern/diff/D22766781)

ann-ss committed:
refactor save_data as non member function

[ghstack-poisoned]

ann-ss committed:
Update on "refactor save_data as non member function"

This PR changes the save_data() member functions of torch::jit::mobile::Module which was introduced in #41403 to be the non member function torch::jit::mobile::_save_parameters() (taking a mobile Module as its first argument).

In addition, this PR:
* adds a getter function _ivalue() for the mobile::Module object
* renames torch::jit::mobile::_load_mobile_data() to torch::jit::mobile_load_parameters()




[ghstack-poisoned]

ann-ss committed:
Update on "refactor save_data as non member function"

This PR changes the save_data() member functions of torch::jit::mobile::Module which was introduced in #41403 to be the non member function torch::jit::mobile::_save_parameters() (taking a mobile Module as its first argument).

In addition, this PR:
* adds a getter function _ivalue() for the mobile::Module object
* renames torch::jit::mobile::_load_mobile_data() to torch::jit::mobile_load_parameters()
* refactors the import.h header file into import.h and import_data.h




[ghstack-poisoned]

iseeyuan reviewed (approved):
LGTM. Thanks for the refactoring!

ann-ss committed:
Update on "refactor save_data as non member function"

This PR changes the save_data() member functions of torch::jit::mobile::Module which was introduced in #41403 to be the non member function torch::jit::mobile::_save_parameters() (taking a mobile Module as its first argument).

In addition, this PR:
* adds a getter function _ivalue() for the mobile::Module object
* renames torch::jit::mobile::_load_mobile_data() to torch::jit::mobile_load_parameters()
* refactors the import.h header file into import.h and import_data.h




[ghstack-poisoned]

ann-ss committed:
Update on "refactor save_data as non member function"

This PR changes the save_data() member functions of torch::jit::mobile::Module which was introduced in #41403 to be the non member function torch::jit::mobile::_save_parameters() (taking a mobile Module as its first argument).

In addition, this PR:
* adds a getter function _ivalue() for the mobile::Module object
* renames torch::jit::mobile::_load_mobile_data() to torch::jit::mobile_load_parameters()
* refactors the import.h header file into import.h and import_data.h


Differential Revision: [D22766781](https://our.internmc.facebook.com/intern/diff/D22766781)

[ghstack-poisoned]

ann-ss committed:
Update on "refactor save_data as non member function"

This PR changes the save_data() member functions of torch::jit::mobile::Module which was introduced in #41403 to be the non member function torch::jit::mobile::_save_parameters() (taking a mobile Module as its first argument).

In addition, this PR:
* adds a getter function _ivalue() for the mobile::Module object
* renames torch::jit::mobile::_load_mobile_data() to torch::jit::mobile_load_parameters()
* refactors the import.h header file into import.h and import_data.h


Differential Revision: [D22766781](https://our.internmc.facebook.com/intern/diff/D22766781)

[ghstack-poisoned]

ann-ss committed:
Update on "refactor save_data as non member function"

This PR changes the save_data() member functions of torch::jit::mobile::Module which was introduced in #41403 to be the non member function torch::jit::mobile::_save_parameters() (taking a mobile Module as its first argument).

In addition, this PR:
* adds a getter function _ivalue() for the mobile::Module object
* renames torch::jit::mobile::_load_mobile_data() to torch::jit::mobile_load_parameters()
* refactors the import.h header file into import.h and import_data.h


Differential Revision: [D22766781](https://our.internmc.facebook.com/intern/diff/D22766781)

[ghstack-poisoned]
